File: slidez-core/src/main/java/com/binarystudio/academy/slidez/domain/poll/dto/PollDto.java
package com.binarystudio.academy.slidez.domain.poll.dto;

import com.binarystudio.academy.slidez.domain.interactive_element.dto.InteractiveElementDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Data
@NoArgsConstructor
@AllArgsConstructor
public class PollDto extends InteractiveElementDto {

	private String title;

	private Boolean isTemplate;

	private Boolean isMulti;

	private List<PollOptionDto> pollOptions;

}

File: slidez-fe-core/src/containers/presentation_session/dto/PollDto.ts
import { PollOptionDto } from './PollOptionDto'
import { InteractiveElement } from './InteractiveElement'
import { PollAnswerDto } from './PollAnswerDto'

export interface PollDto extends InteractiveElement {
    title: string
    isMulti: boolean
    isTemplate: boolean
    options: PollOptionDto[]
    answers: PollAnswerDto[]
}
